### PR TITLE
drivers/clock_control: stm32_common: Fixed PLL configuration 

### PR DESCRIPTION
**Pull request by DanHigginbotham**
-Kept getting an error on STM32g0xx chips that didn't have q-divisor.
-Changed to set prescaler only if it's being used.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this work. Change looks fine, but please fix commit message (which will likely be reported non compliant by testing scripts).
See [upstream contribution guidelines](https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf), slides 21>24

**Comment by erwango** [comment]
Thanks for the commit update. Please also change the commit title (first line) to something like:
`[subsystem]: [summary of change]`

**Comment by Higgy710** [comment]
Still fairly new to git workflow. Thank you all for your patience!

**Comment by str4t0m** [comment]
You again ended up with 3 commits. Could you please drop the 2nd and 3rd one and only keep the first one.

**Comment by Higgy710** [comment]
> You again ended up with 3 commits. Could you please drop the 2nd and 3rd one and only keep the first one.

All fixed!

**Comment by erwango** [comment]
@Higgy710 Thanks for the updates a last mile walk needed:
```
873640af6cb8c674e60d068b2cf3ea49c5120fcf: author email (Dan Higginbotham <47610444+Higgy710@users.noreply.github.com>) must be a real email and cannot end in @users.noreply.github.com
```
This one is tricky as author name is a commit meta information, not visible in commit message (you can see it using `gitk` though). See https://www.git-tower.com/learn/git/faq/change-author-name-email to fix it.

**Comment by erwango** [comment]
Please remove the merge commit.

**Comment by Higgy710** [comment]
I mean to rebase but as I was clicking, it changed to "Update."

**Review by erwango — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Comment by erwango** [comment]
@Higgy710 Now that it was approved, you don't need to update this branch anymore. Each time you rebase, a force push happens, approvals are dismissed automatically and we need to review again.
For this PR to be merged, you need at least 2 approvals from maintainers and then a release engineer will have a look and then merge if eveything is ok. No merge will happen if the approvals are dismissed (by a rebase for instance).
This may take some time, please be patient.

**Comment by Higgy710** [comment]
> @Higgy710 Now that it was approved, you don't need to update this branch anymore. Each time you rebase, a force push happens, approvals are dismissed automatically and we need to review again. For this PR to be merged, you need at least 2 approvals from maintainers and then a release engineer will have a look and then merge if eveything is ok. No merge will happen if the approvals are dismissed (by a rebase for instance). This may take some time, please be patient.

I wasn't sure about that. Thank you for the clarification.

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
